Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m

```diff
@@ -81,23 +77,11 @@ - (instancetype)initWithConversationStyle:(ConversationStyle *)conversationStyle
     return self;
 }
 
-- (void)dealloc
-{
-    [self removeKVOObservers];
-}
-
 - (CGSize)intrinsicContentSize
 {
-    // Since we have `self.autoresizingMask = UIViewAutoresizingFlexibleHeight`, the intrinsicContentSize is used
-    // to determine the height of the rendered inputAccessoryView.
-    CGFloat height = self.toolbarHeight + ConversationInputToolbarBorderViewHeight;
-    if (self.quotedMessagePreview) {
-        height += self.quotedMessageTopMargin;
-        height += self.quotedMessagePreview.intrinsicContentSize.height;
-    }
-    CGSize newSize = CGSizeMake(self.bounds.size.width, height);
-
-    return newSize;
+    // Since we have `self.autoresizingMask = UIViewAutoresizingFlexibleHeight`, we must specify
+    // an intrinsicContentSize. Specifying CGSize.zero causes the height to be determined by autolayout.
```

Comment:
A lot of these changes are around converting the conversation inputToolbar to be self sizing.